nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -0,0 +1,173 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+func resourceNsxtManagerCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtManagerClusterCreate,
+		Read:   resourceNsxtManagerClusterRead,
+		Update: resourceNsxtManagerClusterUpdate,
+		Delete: resourceNsxtManagerClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the NSXT cluster",
+				Computed:    true,
+			},
+			"nodes": {
+				Type:        schema.TypeList,
+				Description: "IP addresses of nodes in the cluster",
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsIPv4Address,
+				},
+				Required: true,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Description: "The authentication password for login",
+				Required:    true,
+				Sensitive:   true,
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Description: "The username for login",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceNsxtManagerClusterCreate(d *schema.ResourceData, m interface{}) error {
+	nodes := getStringListFromSchemaList(d, "nodes")
+	if len(nodes) == 0 {
+		return fmt.Errorf("Please provide at least one node ip")
+	} else if len(nodes) == 1 {
+		log.Printf("[INFO] Only one node ip provided, skip joining nodes process")
```

Comment:
Should the user provide master node in the list?

Reply:
I'm thinking if we want to assume that the host node in terraform provider config will be on this list.
For example nodes    = ["192.168.240.31", "192.168.240.32", "192.168.240.33"]
and in terrraform config we have 
provider "nsxt" {
  host                 = "192.168.240.31"
  username             = "admin"
  password             = "password"
  allow_unverified_ssl = true
  max_retries          = 2
}

Then the master node will be 192.168.240.31

If we want to use the provider to join nodes that does not include the host node for the provider, for example
["192.168.240.32", "192.168.240.33", "192.168.240.34"]
Then we should probably includes an argument master_node in the schema

Reply:
So I think we will be using the host in provider config as master node, in .tf file for the resource we only provide the nodes that will be joining into the cluster